### PR TITLE
Added files required for CRUD operations of schedule invoices

### PR DESCRIPTION
**Pull request by chadivesatish**

### CODE REVIEW

File: models/schedule.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+//schedule Schema :
+var scheduleSchema = mongoose.Schema({
+
+    //Object ID : (Unique ID, automatically created by MongoDb)
+    _id: mongoose.Schema.Types.ObjectId,
+
+    //Validation for client :
+    client: {
```
Comment (faisal3389) [inline models/schedule.js:13]: let make this `clientId`

File: routes/schedule.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+//Middleware's Imported :
+var SF_Pag = require("../middlewares/search_functionality-Pagination");          //Middleware for Search-Functionality and Pagination
+
+
+//Validations Imported :
+var Schedule_Validator = require("../validations/schedule_validations");
+
+
+// imported check authorizations :
+//var Admin_Authorization = require("../check_authorization/admin_authorization");
```
Comment (faisal3389) [inline routes/schedule.js:20]: remove the commented code

File: models/schedule.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    //Validation for time :
+    time: {
+        type: String,
+        required: [true, "please enter time"]
+    },
+    isActive:{
+        type: Boolean,
+        default:true
+    }
```
Comment (faisal3389) [inline models/schedule.js:44]: use `isDisabled` key here

File: routes/schedule.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+//GET Request for schedule ID :
+router.get("/:schedule_id", async (req, res) => {
+
+    try {
+
+        //Finding schedule by ID :
+        const id = req.params.schedule_id;
+        const schedule = await Schedule.findOne({ _id: id, isActive: true });
+
+        if (schedule == null) {
```
Comment (faisal3389) [inline routes/schedule.js:110]: use `if (!schedule) {`

File: routes/schedule.js [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+                    "code": 404,
+                    "message": constants.MODEL_NOT_FOUND
+                }
+            });
+        } else {
+
+            const { is_active, client , invoiceNumber, date, frequency, time } = req.body;
+
+            //Updating schedule :
+            schedule.is_active = is_active;
```
Comment (faisal3389) [inline routes/schedule.js:186]: use a new variable to modify data, 
`updatedSchedule = {... schedule}`

File: routes/schedule.js [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+            res.status(404).json({
+                "status": {
+                    "success": false,
+                    "code": 404,
+                    "message": constants.MODEL_NOT_FOUND
+                }
+            });
+        } else {
+
+            const { is_active, client , invoiceNumber, date, frequency, time } = req.body;
```
Comment (faisal3389) [inline routes/schedule.js:183]: use camel case: `isActive`

File: routes/schedule.js (hunk 5 above)
Comment (faisal3389) [inline routes/schedule.js:186]: use `schedule.is_active = isActive;`

File: routes/schedule.js [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+//DELETE Request for schedule ID :
+router.delete("/:schedule_id", async (req, res) => {
+
+    try {
+
+        //Finding schedule :
+        const id = req.params.schedule_id;
+        const schedule = await Schedule.findById(id);
+
+        if (schedule == null) {
```
Comment (faisal3389) [inline routes/schedule.js:227]: same here

File: routes/schedule.js [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+var Schedule_Validator = require("../validations/schedule_validations");
+
+
+// imported check authorizations :
+//var Admin_Authorization = require("../check_authorization/admin_authorization");
+
+
+//Importing Constants :
+var constants_function = require("../constants/constants");
+var constants = constants_function("schedule");
```
[the remark this answers is not in the document]
Comment (chadivesatish) [reply routes/schedule.js:25]: those are required to return some standard response messages

File: routes/schedule.js (hunk 5 above)
Comment (chadivesatish) [reply routes/schedule.js:186]: schedule itself is a new variable to modify original data from database

File: package.json [hunk 9]
```diff
@@ -20,6 +20,7 @@
     "jsonwebtoken": "^8.5.1",
     "mongoose": "^5.13.2",
     "morgan": "~1.9.1",
+    "node-cron": "^3.0.1",
```
Comment (faisal3389) [inline package.json:23]: remove this

File: routes/schedule.js [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+                "code": 400,
+                "message": errors.array()[0].msg
+            }
+        });
+    }
+    try {
+        const { isDisabled, clientId, scheduleName, invoiceNumber, date, frequency, time } = req.body;
+        const schedule = new Schedule({
+            _id: new mongoose.Types.ObjectId(),
+            isDisabled, clientId,scheduleName, invoiceNumber, date, frequency, time
```
Comment (faisal3389) [inline routes/schedule.js:39]: no scheduleName

**Pull request merged**